modules/apps/portal-crypto-hash/portal-crypto-hash-provider/portal-crypto-hash-provider-bcrypt/src/main/java/com/liferay/portal/crypto/hash/provider/bcrypt/internal/BCryptCryptoHashProvider.java

```diff
@@ -19,36 +19,49 @@
 
 import java.nio.charset.StandardCharsets;
 
+import java.util.Map;
+
 import jodd.crypt.BCrypt;
 
 /**
  * @author Arthur Chan
  */
 public class BCryptCryptoHashProvider implements CryptoHashProvider {
 
+	public BCryptCryptoHashProvider(String cryptoHashProviderFactoryName) {
+		_cryptoHashProviderFactoryName = cryptoHashProviderFactoryName;
+
+		_rounds = 10;
+	}
+
+	public BCryptCryptoHashProvider(
+		String cryptoHashProviderFactoryName,
+		Map<String, ?> cryptoHashProviderProperties) {
+
+		_cryptoHashProviderFactoryName = cryptoHashProviderFactoryName;
+
+		_rounds = (Integer)cryptoHashProviderProperties.get("rounds");
```

Comment:
should we extract these properties to constants?

Reply:
also please use `MapUtil`

Reply:
> should we extract these properties to constants?

Do you mean something like
` Public static final String ROUNDS = "rounds"?`

Where do we put them? In some constantKey class?

How do users get a hold of these keys when trying to create provider instance?

Reply:
Done:
https://github.com/liferay-appsec/liferay-portal/pull/294/commits/4a76d7ac40791e525c1c1ebe8eb54eb927dce0ef#diff-6ffb8a56521492ce2a43aeb57b27302b99cca221952837c24ab52c1060f07cf9R33

https://github.com/liferay-appsec/liferay-portal/pull/294/commits/4a76d7ac40791e525c1c1ebe8eb54eb927dce0ef#diff-6ffb8a56521492ce2a43aeb57b27302b99cca221952837c24ab52c1060f07cf9R51